### ISSUE TITLE
T-9.3: RL Training Pipeline (PPO/TD3)

### ISSUE THREAD

**Issue by kipotz1986** [kipotz1986/apex-trading-bot#65]
**Labels:** `epic-9`, `self-learning`, `reinforcement-learning`, `priority-critical`
**Milestone:** Fase 4 — Intelligence
**Depends On:** T-9.1, T-9.2

### Deskripsi
Implementasi training pipeline menggunakan Proximal Policy Optimization (PPO). Bot belajar dari pengalaman trading-nya untuk memperbaiki keputusan di masa depan.

### Apa itu PPO?
PPO adalah algoritma RL yang belajar dari trial-and-error. Bayangkan anak kecil belajar berjalan: jatuh → sakit (negative reward) → coba cara lain → berhasil → senang (positive reward). PPO melakukan ini secara matematis dan cepat.

#### 1. Install dependency
```bash
pip install stable-baselines3 gymnasium
echo "stable-baselines3>=2.3.0" >> backend/requirements.txt
echo "gymnasium>=0.29.0" >> backend/requirements.txt
```

#### 2. Buat custom Gym environment
```python
# backend/app/services/learning/trading_env.py

import gymnasium as gym
import numpy as np
from gymnasium import spaces


class TradingEnvironment(gym.Env):
    """
    Custom Gym environment untuk training RL agent.
    
    State: Market indicators + portfolio state (dari T-9.2)
    Actions: BUY, SELL, HOLD, INCREASE, DECREASE
    Reward: Risk-adjusted PnL (dari T-9.1)
    """
    
    def __init__(self, historical_data, initial_balance=10000):
        super().__init__()
        
        self.data = historical_data
        self.initial_balance = initial_balance
        
        # Action space: 5 discrete actions
        self.action_space = spaces.Discrete(5)
        # 0=HOLD, 1=BUY, 2=SELL, 3=INCREASE, 4=DECREASE
        
        # Observation space: state vector dimension
        self.observation_space = spaces.Box(
            low=-1, high=1,
            shape=(25,),  # 25 features
            dtype=np.float32
        )
        
        self.reset()
    
    def reset(self, seed=None, options=None):
        self.current_step = 0
        self.balance = self.initial_balance
        self.position = None
        self.total_reward = 0
        return self._get_state(), {}
    
    def step(self, action):
        # 1. Execute action
        # 2. Calculate reward
        # 3. Check if done
        # 4. Return new state, reward, done, info
        ...
```

#### 3. Training script
```python
from stable_baselines3 import PPO

# Create environment
env = TradingEnvironment(historical_data)

# Create model
model = PPO(
    "MlpPolicy",
    env,
    verbose=1,
    learning_rate=3e-4,
    n_steps=2048,
    batch_size=64,
    n_epochs=10,
)

# Train
model.learn(total_timesteps=100_000)

# Save
model.save("models/apex_rl_v1")
```

### Definition of Done
- [ ] Custom Gym environment berfungsi
- [ ] PPO model bisa di-train tanpa error
- [ ] Model bisa di-save dan di-load
- [ ] Training reward meningkat seiring waktu (learning curve positif)
- [ ] Model bisa menghasilkan action dari state

### File yang Dibuat
- `[NEW]` `backend/app/services/learning/trading_env.py`
- `[NEW]` `backend/app/services/learning/trainer.py`

---
